pennylane/pulse/hardware_hamiltonian.py

```diff
@@ -25,35 +25,36 @@
 
 from .parametrized_hamiltonian import ParametrizedHamiltonian
 
+# from .transmon import TransmonSettings
+# from .rydberg import RydbergSettings
+
 
-def drive(amplitude, phase, detuning, wires):
+def drive(amplitude, phase, wires):
     r"""Constructs a :class:`ParametrizedHamiltonian` representing the action of a driving electromagnetic
     field with a qubit.
 
     .. math::
-        \frac{1}{2} \sum_{j \in \text{wires}} \Omega(t) \left(e^{i \phi(t)} \sigma^+_j + e^{-i \phi(t)} \sigma^-_j \right) -
-        \Delta(t) \sigma^z_j
+        \frac{1}{2} \sum_{j \in \text{wires}} \Omega(t) \left(e^{i \phi(t)} \sigma^+_j + e^{-i \phi(t)} \sigma^-_j \right)
 
-    where :math:`\Omega`, :math:`\phi` and :math:`\Delta` correspond to the amplitude, phase and detuning of the electromagnetic
-    driving field and :math:`j` corresponds to the wire index. We are describing the Hamiltonian in terms of ladder operators
-    :math:`\sigma^\pm = \frac{1}{2}(\sigma_x \pm i \sigma_y)`.
-    Note that the detuning :math:`\Delta := \omega_q - \nu` is defined as the difference between the qubit frequency :math:`\omega_q`
-    and the electromagntic field driving frequency :math:`\nu`. For more details, see the theoretical background section below.
+    where :math:`\Omega` and :math:`\phi` correspond to the amplitude and phase of the
+    electromagnetic driving field and :math:`j` corresponds to the wire index. We are describing the Hamiltonian
+    in terms of ladder operators :math:`\sigma^\pm = \frac{1}{2}(\sigma_x \pm i \sigma_y)`. Note that it is common
+    to describe the exponent of the phase factor as :math:`\exp(i(\phi(t) + \nu t))`, where :math:`\nu` is the
+    drive frequency (see theoretical background below).
```

Comment:
I think we do need to mention something about frequency, because otherwise this is confusing to anyone who is used to the formulation Korbinian references here, but I might suggest something more like:

"Note that this function corresponds to an on-resonance drive (frequency = $\omega_q$). A separate term accounting for the effect of frequency detuning can be added. For details on frequency and off-resonance drives, see theoretical background."

That addresses the concern of anyone who wants to know where frequency went without going into any detail about alternate formulations.

Reply:
> "Note that this function corresponds to an on-resonance drive (frequency = ). A separate term accounting for the effect of frequency detuning can be added. For details on frequency and off-resonance drives, see theoretical background."

To me this would be more confusing 😬 I know _now_ that with the transformation going to the rotating frame this frequency term and the $\omega a^\dagger a$ term cancel when on resonance, but I dont think this is common knowledge.

The argument for noting this here is just to make it clear that there are different conventions and that their relations are explained in the theoretical background section below. I expanded that this is a matter of convention depending on the sub-field:

> Note that depending on the
    hardware realization (neutral atoms, superconducting qubits), there are different conventions and notations.
    E.g., for superconducting qubits it is common to describe the exponent of the phase factor as :math:`\exp(i(\phi(t) + \nu t))`, where :math:`\nu` is the
    drive frequency. We describe their relations in the theoretical background section below.

Reply:
For anyone used to working with hardware, I think it will go a long way to explain how you could have a hamiltonian for that doesn't include drive frequency, and is still applicable to transmons. It would definitely have made sense to me (not mathematically, but conceptually) before going into all of this. It explains that the hamiltonian is expressed in terms of detuning, and detuning is 0, so its not that frequency is missing, its just on-resonance. These are the concepts people are used to working with when driving transmons on hardware. 

Without the word "on-resonance" in there somewhere, I don't think I ever would have on my own (without going through the extensive reading and discussion) understood how this could apply to a transom system.  I don't know about people who are only used to working with the formulas for transmons and not the hardware. 

But I think this confusion does raise the question - do we really want a "generalized" drive function that is user-facing? I'm not even sure we're going to end up being able to use it for the hardware transmons, because I think we will need to include all 3 (frequency, amplitude and phase) in a single callable, which won't work within the structure of AmplitudeAndPhase. 

But I propose we relegate this discussion to a follow-up PR that "cleans up the docs" like last release, so we can get this merged.

Reply:
> But I propose we relegate this discussion to a follow-up PR that "cleans up the docs" like last release, so we can get this merged.

sounds good 👍